src/py/flwr/server/driver/grpc_driver.py

```diff
@@ -259,7 +299,7 @@ def send_and_receive(
         end_time = time.time() + (timeout if timeout is not None else 0.0)
         ret: list[Message] = []
         while timeout is None or time.time() < end_time:
-            res_msgs = self.pull_messages(msg_ids)
+            res_msgs = list(self.pull_messages(msg_ids))
```

Comment:
We don't need this here. `list.extend` accepts an iterable.
```suggestion
            res_msgs = self.pull_messages(msg_ids)
```

Reply:
For example:
![image](https://github.com/user-attachments/assets/16a99398-beb5-40c2-bfff-4ac8031c84d3)

Reply:
Keeping the `list`because `res_msgs` is used later on line 303.